client/app/scripts/reducers/root.js

```diff
@@ -721,6 +721,12 @@ export function rootReducer(state = initialState, action) {
       return state.set('showingTroubleshootingMenu', !state.get('showingTroubleshootingMenu'));
     }
 
+    case ActionTypes.CHANGE_INSTANCE: {
+      state = state.set('selectedNodeId', null);
+      state = state.update('nodeDetails', nodes => nodes.clear());
```

Comment:
I think that `controlPipes` should also be cleared here, just by seeing what `closeNodeDetails` does. And then I would actually replace the content of `closeAllNodeDetails` helper with these lines and just use it in this action, as it's all more straightforward and consistent that way.